### PR TITLE
feat: security migration

### PR DESCRIPTION
**Pull request by arunjaindev**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

## Checklist

- [ ] The title of the PR states what changed and the related issues number (used for the release note).
- [ ] Does this PR require documentation updates?
- [ ] I've updated documentation as required by this PR.
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/Shared/Components/ImageCardAccordion/types.ts [hunk 1]
```diff
@@ -27,6 +28,8 @@ export interface ImageCardAccordionProps extends MaterialSecurityInfoType {
     isScanned: boolean
     isScanEnabled: boolean
     isScanV2Enabled: boolean
+    SecurityModalSidebar: React.FC<SidebarPropsType>
+    getSecurityScan: ({ appId, envId, artifactId }: AppDetailsPayload) => Promise<ResponseType<ApiResponseResultType>>
```
Comment (AbhishekA1509) [inline src/Shared/Components/ImageCardAccordion/types.ts:32]: Pick<AppDetailsPayload, 'appId' | 'envId' | 'artifactId'>

File: src/Shared/Components/Security/Vulnerabilities/types.ts [hunk 2]
```diff
@@ -37,7 +41,7 @@ export interface UseGetSecurityVulnerabilitiesProps {
 
 export interface UseGetSecurityVulnerabilitiesReturnType {
     scanDetailsLoading: boolean
-    scanResultResponse: Awaited<ReturnType<typeof getSecurityScan>>
+    scanResultResponse: ResponseType<ApiResponseResultType>
```
Comment (AbhishekA1509) [inline src/Shared/Components/Security/Vulnerabilities/types.ts:44]: Suggestion: Could have made typing more generic for hooks

File: src/Shared/Components/Security/SecurityModal/types.ts [hunk 3]
```diff
@@ -271,6 +246,19 @@ export type ApiResponseResultType = {
     [CATEGORIES.KUBERNETES_MANIFEST]: KubernetesManifest
 }
 
+interface SecurityModalBaseProps extends Partial<Pick<SidebarPropsType, 'isExternalCI'>> {
+    isLoading: boolean
+    error: any
```
Comment (AbhishekA1509) [inline src/Shared/Components/Security/SecurityModal/types.ts:251]: Can use ServerErrors

File: src/Shared/Components/Security/SecurityModal/service.ts [hunk 4]
```diff
@@ -2,90 +2,31 @@
...
+        lastExecution: response.result.executionTime,
+        objectType: response.result.objectType,
+        vulnerabilities: response.result.vulnerabilities?.map((cve) => ({
+            name: cve.cveName,
+            severity: cve.severity,
+            package: cve.package,
+            version: cve.currentVersion,
+            fixedVersion: cve.fixedVersion,
+            policy: cve.permission,
+        })),
```
Comment (AbhishekA1509) [inline src/Shared/Components/Security/SecurityModal/service.ts:29]: Please add null checks and fallbacks

**Review by AbhishekA1509 — APPROVED** [review]

**Review by AbhishekA1509 — APPROVED** [review]

**Review by vivek-devtron — APPROVED** [review]

**Pull request merged**